From fb0de11b283a6bc88b1fbdecfa464eeb34dbf0d8 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 4 May 2017 16:02:20 +0300
Subject: [PATCH 098/767] staging: fsl-dpaa2/eth: Change link settings while
 interface up

Starting with MC 10.1.1, DPNI link options can be changed
while the interface is up. Remove the check.

While here, remove an unnecessary debug message.

TODO: Read DPNI API version and do this conditionally

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
index 5a976d3016b5..381a3ab897ff 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-ethtool.c
@@ -133,18 +133,6 @@ dpaa2_eth_set_link_ksettings(struct net_device *net_dev,
 	struct dpni_link_cfg cfg = {0};
 	int err = 0;
 
-	/* If using an older MC version, the DPNI must be down
-	 * in order to be able to change link settings. Taking steps to let
-	 * the user know that.
-	 */
-	if (dpaa2_eth_cmp_dpni_ver(priv, DPNI_DYNAMIC_LINK_SET_VER_MAJOR,
-				   DPNI_DYNAMIC_LINK_SET_VER_MINOR) < 0) {
-		if (netif_running(net_dev)) {
-			netdev_info(net_dev, "Interface must be brought down first.\n");
-			return -EACCES;
-		}
-	}
-
 	/* Need to interrogate link state to get flow control params */
 	err = dpni_get_link_state(priv->mc_io, 0, priv->mc_token, &state);
 	if (err) {
-- 
2.17.0

